woocommerce/woocommerce-admin
# Navigation: Reset submenu before making Flyout

Fixes https://github.com/woocommerce/woocommerce-admin/issues/6377

Some automatically migrated menu items are still included in the WooCommerce Submenu, so they are also added to the flyout. This PR makes sure all items have the `hide-if-js` before constructing the flyout.

@joshuatf This works, but I'm not sure its the best way. I'm not fully understanding how MailChimp gets migrated, but it seems that would be a better time to add the `hide-if-js` class than what I've done. Also, just to note this solution results in the class added twice if its already present, which I'll fixup if this is the direction.

### Detailed test instructions:

1. Activate the MailChimp plugin
2. Turn on Navigation
3. See the flyout on the WP screen and MailChimp should not be included.
